src/NonEmptyArray.ts

```diff
@@ -289,6 +290,35 @@ export const getSemigroup = <A = never>(): Semigroup<NonEmptyArray<A>> => {
   return { concat }
 }
 
+/**
+ *  Group equal, consecutive elements of an array into arrays.
+ */
+export const group = <A>(S: Setoid<A>) => (as: Array<A>): Array<NonEmptyArray<A>> =>
```

Comment:
Nice. However for a core API like this I'd go for a more efficient mutable solution, something along the lines of

```ts
export const group2 = <A>(S: Setoid<A>) => (as: Array<A>): Array<NonEmptyArray<A>> => {
  const r: Array<NonEmptyArray<A>> = []
  const len = as.length
  if (len === 0) {
    return r
  }
  let a: A = as[0]
  let nonEmpty: Array<A> = [a]
  for (let i = 1; i < as.length; i++) {
    const x = as[i]
    if (S.equals(x, a)) {
      nonEmpty.push(x)
    } else {
      r.push(unsafeFromArray(nonEmpty))
      nonEmpty = [x]
    }
    a = x
  }
  r.push(unsafeFromArray(nonEmpty))
  return r
}
```

Quick benchmark (node v10.1.0)

```
group x 629,882 ops/sec ±0.82% (73 runs sampled)
group2 x 4,459,547 ops/sec ±0.77% (87 runs sampled)
```